Lib/argparse.py

```diff
@@ -2070,37 +2069,29 @@ def consume_positionals(start_index):
         # passed the last option string
         extras = []
         start_index = 0
-        if option_string_indices:
-            max_option_string_index = max(option_string_indices)
-        else:
-            max_option_string_index = -1
-        while start_index <= max_option_string_index:
+
+        for option_tuple_index, option_tuple in option_tuples:
+            if start_index > option_tuple_index:
```

Comment:
In the old code. This passes L2084.

In this particular case will L2098 then always be true? If so, should we also re-assign `start_index`.
If not I think this code path would go on until L2013 without any other `continues`. Is this a change in behaviour? Or is this simply never not true?

Reply:
Thanks for the clear explanation!

One follow-up. My initial point was about case 4. In that case, does old L2090 not reassign `start_index`? The continue on new L2071 doesn't reassign, which is why I thought there might be a subtle difference.

Reply:
> My initial point was about case 4. In that case, does old L2090 not reassign start_index? The continue on new L2071 doesn't reassign, which is why I thought there might be a subtle difference.

The new code reassigns `start_index` on L2075. Basically the old and new code maintain the same invariant - `start_index` will be set to the value returned by `consume_positionals()`. It is just the old code passes the same value through `positionals_end_index` first.

The continue at L2071 does not need to reassign `start_index` because at this point the code just tries to rewind `option_tuples` to the next index that is `>= start_index`.

Reply:
Ah yeah. Wow, for some reason I couldn't wrap my around that. Sorry for making such a point about this, I should have understood this. Thanks for the careful explanation anyway!